### PR TITLE
Add optional check for current memory usage to abort the process

### PR DESCRIPTION
**Pull request by davidbanham**
Better than getting OOMed!

#### Description

Provide a way to take down the process on our own terms if memory usage grows beyond a bound.

----
#### Changes
Adds an env var for max memory size.
Checks that env var against current RSS
If the limit is exceeded, the process is immediately exited and a core dump is generated for later analysis

----

#### Testing Instructions
npm test
And then to see it in action:
STEVEO_MAX_RSS=1 npm test

### CODE REVIEW

File: src/runner.js [hunk 1]
```diff
@@ -17,6 +20,13 @@ const Runner = (config: Config, registry: Reg, logger: Object) => {
     for (const m of messages) { // eslint-disable-line
       let params = null;
       try {
+        if ( maxRss ) {
+          const currentRss = memoryUsage().rss;
+          if (currentRss > maxRss) {
```
Comment (danielmoi) [inline src/runner.js]: `>=` ?

File: src/runner.js [hunk 2]
```diff
@@ -17,6 +20,13 @@ const Runner = (config: Config, registry: Reg, logger: Object) => {
...
+            logger.error(`Steveo - Memory ${currentRss} is above max ${maxRss}. Killing the process`);
```
Comment (danielmoi) [inline src/runner.js]: is there / is it worth logging the process id?

Comment (davidbanham) [inline src/runner.js]: Nah, I don't think the PID really matters to us. It'd be the container/task ID we'd really be interested in anyway. Would be super nice to add those fields to the logs that end up in Loggly as an aside.

File: src/runner.js (hunk 1 above)
Comment (davidbanham) [inline src/runner.js]: Maybe? I dunno. Kinda subjective I think. Semantically I think "If it's at the max, that's cool. If it's beyond the max, oh no cover your ears."

If you think it'd be clearer as >= though I'm very happy with that. Whatever the majority view is we should go with.

**Pull request closed without merging**